diff -urN cone-0.69.orig/cone/addressbook.C cone-0.69/cone/addressbook.C
--- cone-0.69.orig/cone/addressbook.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/addressbook.C	2007-04-06 22:31:41.000000000 +0530
@@ -45,6 +45,7 @@
 #include <list>
 #include <set>
 #include <map>
+#include <algorithm>
 
 #include "addressbookinterfacemail.H"
 #include "addressbookinterfaceldap.H"
diff -urN cone-0.69.orig/cone/configscreen.C cone-0.69/cone/configscreen.C
--- cone-0.69.orig/cone/configscreen.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/configscreen.C	2007-04-06 22:31:41.000000000 +0530
@@ -30,6 +30,7 @@
 
 #include <sstream>
 #include <errno.h>
+#include <algorithm>
 
 extern struct CustomColor color_misc_promptColor;
 extern struct CustomColor color_misc_inputField;
diff -urN cone-0.69.orig/cone/curseshierarchy.C cone-0.69/cone/curseshierarchy.C
--- cone-0.69.orig/cone/curseshierarchy.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/curseshierarchy.C	2007-04-06 22:31:41.000000000 +0530
@@ -32,6 +32,7 @@
 #include "acl.H"
 #include <errno.h>
 #include <set>
+#include <algorithm>
 
 using namespace std;
 
diff -urN cone-0.69.orig/cone/gpg.C cone-0.69/cone/gpg.C
--- cone-0.69.orig/cone/gpg.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/gpg.C	2007-04-06 22:31:41.000000000 +0530
@@ -21,6 +21,7 @@
 #include "wraptext.H"
 #include <vector>
 #include <set>
+#include <algorithm>
 #include <errno.h>
 
 extern char ucheck[];
diff -urN cone-0.69.orig/cone/hierarchy.C cone-0.69/cone/hierarchy.C
--- cone-0.69.orig/cone/hierarchy.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/hierarchy.C	2007-04-06 22:31:41.000000000 +0530
@@ -16,6 +16,7 @@
 #include "libmail/mail.H"
 
 #include <queue>
+#include <algorithm>
 
 using namespace std;
 
diff -urN cone-0.69.orig/cone/init.C cone-0.69/cone/init.C
--- cone-0.69.orig/cone/init.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/init.C	2007-04-06 22:31:41.000000000 +0530
@@ -28,6 +28,7 @@
 #include "messagesize.H"
 #include <string>
 #include <iostream>
+#include <climits>
 
 #include "init.H"
 #include "buildversion.H"
diff -urN cone-0.69.orig/cone/myfolder.C cone-0.69/cone/myfolder.C
--- cone-0.69.orig/cone/myfolder.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/myfolder.C	2007-04-06 22:31:41.000000000 +0530
@@ -24,6 +24,7 @@
 
 #include <errno.h>
 
+#include <algorithm>
 #include <sstream>
 #include <fstream>
 #include <iomanip>
diff -urN cone-0.69.orig/cone/mymessage.C cone-0.69/cone/mymessage.C
--- cone-0.69.orig/cone/mymessage.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/mymessage.C	2007-04-06 22:31:41.000000000 +0530
@@ -53,6 +53,7 @@
 #include <iomanip>
 #include <fstream>
 #include <iostream>
+#include <algorithm>
 
 using namespace std;
 
diff -urN cone-0.69.orig/cone/nntpcommand.C cone-0.69/cone/nntpcommand.C
--- cone-0.69.orig/cone/nntpcommand.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/cone/nntpcommand.C	2007-04-06 22:31:41.000000000 +0530
@@ -11,6 +11,7 @@
 #include "gettext.H"
 #include <errno.h>
 #include <stdio.h>
+#include <cstdlib>
 
 #include <sys/types.h>
 #if HAVE_UNISTD_H
diff -urN cone-0.69.orig/libmail/imap.C cone-0.69/libmail/imap.C
--- cone-0.69.orig/libmail/imap.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/libmail/imap.C	2007-04-06 22:31:41.000000000 +0530
@@ -21,6 +21,7 @@
 #include <unicode/unicode.h>
 #include <sstream>
 #include <iomanip>
+#include <algorithm>
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
diff -urN cone-0.69.orig/libmail/imaplogin.C cone-0.69/libmail/imaplogin.C
--- cone-0.69.orig/libmail/imaplogin.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/libmail/imaplogin.C	2007-04-06 22:31:41.000000000 +0530
@@ -17,6 +17,7 @@
 #include <errno.h>
 #include <stdio.h>
 
+#include <algorithm>
 #include <iostream>
 #include <iomanip>
 #include <sstream>
diff -urN cone-0.69.orig/libmail/mailtool.C cone-0.69/libmail/mailtool.C
--- cone-0.69.orig/libmail/mailtool.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/libmail/mailtool.C	2007-04-06 22:31:41.000000000 +0530
@@ -23,6 +23,7 @@
 
 #include <iostream>
 #include <iomanip>
+#include <algorithm>
 #include <stdlib.h>
 #include <stdio.h>
 #include <ctype.h>
diff -urN cone-0.69.orig/libmail/nntpxover.C cone-0.69/libmail/nntpxover.C
--- cone-0.69.orig/libmail/nntpxover.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/libmail/nntpxover.C	2007-04-06 22:31:41.000000000 +0530
@@ -9,6 +9,7 @@
 #include "generic.H"
 #include "envelope.H"
 
+#include <algorithm>
 #include <sstream>
 
 using namespace std;
diff -urN cone-0.69.orig/libmail/rfc2047encode.C cone-0.69/libmail/rfc2047encode.C
--- cone-0.69.orig/libmail/rfc2047encode.C	2007-04-06 22:31:30.000000000 +0530
+++ cone-0.69/libmail/rfc2047encode.C	2007-04-06 22:31:41.000000000 +0530
@@ -9,6 +9,8 @@
 #include "rfc822/rfc2047.h"
 #include "mail.H"
 
+#include <cstdlib>
+
 mail::rfc2047::encode::encode(std::string txt, std::string charset)
 {
 	char *p=rfc2047_encode_str(txt.c_str(), charset.c_str(),
