Index: cone-0.74/libmail/configure
===================================================================
--- cone-0.74.orig/libmail/configure	2008-04-25 11:41:49.000000000 +0530
+++ cone-0.74/libmail/configure	2008-04-25 11:41:58.000000000 +0530
@@ -19397,7 +19397,7 @@
 done
 IFS=$as_save_IFS
 
-  test -z "$ac_cv_path_SENDMAIL" && ac_cv_path_SENDMAIL="/usr/bin/sendmail"
+  test -z "$ac_cv_path_SENDMAIL" && ac_cv_path_SENDMAIL="/usr/sbin/sendmail"
   ;;
 esac
 fi
Index: cone-0.74/libmail/configure.in
===================================================================
--- cone-0.74.orig/libmail/configure.in	2008-04-25 11:42:23.000000000 +0530
+++ cone-0.74/libmail/configure.in	2008-04-25 11:42:31.000000000 +0530
@@ -23,7 +23,7 @@
 
 LPATH="$PATH:/usr/local/bin:/usr/sbin:/sbin:/lib"
 
-AC_PATH_PROG(SENDMAIL, sendmail, /usr/bin/sendmail, $LPATH)
+AC_PATH_PROG(SENDMAIL, sendmail, /usr/sbin/sendmail, $LPATH)
 
 AC_DEFINE_UNQUOTED(SENDMAIL, "$SENDMAIL",
 			     [ Local sendmail program ])
